prototype/mobile_endpoint/dao.py

```diff
@@ -0,0 +1,238 @@
+from abc import ABCMeta, abstractmethod
+import functools
+import types
+from uuid import UUID
+
+import redis
+from sqlalchemy.orm import contains_eager, defer
+from mobile_endpoint.case.models import CommCareCase
+
+from mobile_endpoint.exceptions import IllegalCaseId
+from mobile_endpoint.form.models import XFormInstance
+from mobile_endpoint.models import db, Synclog, FormData, CaseData, CaseIndex
+from mobile_endpoint.utils import chunked, get_with_lock
+
+
+def to_generic(fn):
```

Comment:
seems like maybe just a convenience way of auto-converting return values rather than having to convert explicitly inside the function?